docker-compose/send-sample-metrics.py

```diff
@@ -0,0 +1,114 @@
+    num_samples = 250    # Minimum of 80 events need to be in Kafka topic to see output
+    print("Sending", num_samples, "random sample messages to Kafka...")
+
+    producer = KafkaProducer(bootstrap_servers=kafka_boostrap_servers)
+    for i in range(0, num_samples):
+        send_msgpack_to_kafka(producer, topic='aa-metrics') # Send to kafka using msgpack format
+        # send_json_to_kafka()              # Send to kafka using JSON format
+    producer.flush()
+
+    print("Checking for anomalies from Kafka...\n")
+    weak_anomaly_values, strong_anomaly_values = consume_from_kafka('anomalies')
+    print("\n-- Found", len(weak_anomaly_values), "WEAK Anomalies")
+    print("-- Weak Anomalous values:", weak_anomaly_values)
+    print("\n-- Found", len(strong_anomaly_values), "STRONG Anomalies")
+    print("-- Strong Anomalous values:", strong_anomaly_values)
+    
+
+# Default Msgpack Serde - Use if ad-mapper.conf is set to "com.expedia.adaptivealerting.kafka.serde.MetricDataMessagePackSerde"
+def send_msgpack_to_kafka(producer, topic):
+    metric_data = {
+        "OrgId": 1, 
+        "Name": "samplemetric", 
+        "Interval": 15,
+        "Value": round(random.uniform(10,200), 1),
+        "Unit": "unknown",
+        "Time": int(time.time()),
+        "Mtype": "gauge",
+        "Tags": ["what=samplemetric", "m_application=sample-web", "region=primary", "env=prod"]
+    }
+    packed_metric_data = msgpack.packb(metric_data, use_bin_type=True)
+    producer.send(topic, key=b'prod.primary.sample-web.sample-metric', value=packed_metric_data)
+
+
+# JSON Serde - Use if ad-mapper.conf is set to "com.expedia.adaptivealerting.kafka.serde.MetricDataJsonSerde"
+def send_json_to_kafka(producer, topic):
+    metric_data = {
+                    "metricDefinition": {
+                        "key": "samplemetric",
+                        "tags": {
+                            "kv": {
+                                "what": "samplemetric",
+                                "m_application": "sample-web",
+                                "region": "primary",
+                                "env": "prod"
+                            },
+                            "v": []
+                        },
+                        "meta": {
+                            "kv": {},
+                            "v": []
+                        }
+                    },
+                    "value": round(random.uniform(10,200), 1),
+                    "timestamp": int(time.time())
+                }
+    metric_data = json.dumps(metric_data).encode()
+    producer.send(topic, key=b'prod.primary.sample-web.sample-metric', value=metric_data)
+
+
+def consume_from_kafka(topic):
+    output_count = 0
+    anomaly_count = 0
+    strong_values = []
+    weak_values = []
+    consumer = KafkaConsumer(topic,
+                            bootstrap_servers=kafka_boostrap_servers, 
+                            auto_offset_reset='earliest', 
```

Comment:
don't we want to use latest?

Reply:
I can do latest, but I will need to run the consumer function in a separate thread since I have to be simultaneously reading from the topic while metrics are being sent in. Otherwise the topic will be empty.